py/pycompiler.vim

```diff
@@ -747,6 +746,8 @@ function s:PythonCompiler.compile_identifier(node)
     let name = 'vval'
   elseif name =~ '^[sa]:'
     let name = name[2:]
+  elseif index(s:reserved_keywords, name) >= 0
```

Comment:
If the argument is `a:lambda`, it still results in `lambda`. (see above condition).

We can add `_` check at the end of function, instead of `elseif`.